src/formatDate.js

```diff
@@ -8,7 +8,34 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const year4 = 'YYYY';
+  const year2 = 'YY';
+
+  const separatedDate = date.split(fromFormat[3]);
+
+  const dateData = {};
+
+  for (let i = 0; i < separatedDate.length; i++) {
+    dateData[fromFormat[i]] = separatedDate[i];
+  }
+
+  if (!dateData[year4]) {
+    if (dateData[year2] < 30) {
+      dateData[year4] = `20${dateData[year2]}`;
+    } else {
+      dateData[year4] = `19${dateData[year2]}`;
+    }
+  } else {
+    dateData[year2] = dateData[year4].slice(2, 4);
+  }
```

Comment:
The condition here assumes that if 'YYYY' is not present in dateData, 'YY' must be present. However, this might not be the case if 'fromFormat' does not include 'YY' or 'YYYY'. It would be safer to check explicitly for 'YY' in dateData before attempting to convert it to 'YYYY'.